scripts/system_performance/system_performance_queries.py

```diff
@@ -0,0 +1,156 @@
+# ############################################################################### #
+# Autoreduction Repository : https://github.com/ISISScientificComputing/autoreduce
+#
+# Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI
+# SPDX - License - Identifier: GPL-3.0-or-later
+# ############################################################################### #
+
+"""
+The purpose of this script is for performing MySQL queries to monitor system state
+performance and health.
+"""
+from __future__ import print_function
+import itertools
+import logging
+from utils.clients.database_client import DatabaseClient
+
+
+class DatabaseMonitorChecks(object):
+    """Class to query system performance"""
+    table = "reduction_viewer_reductionrun"
+
+    # Establishing a connection with Database using Database Client
+    def __init__(self):
+        self.database = DatabaseClient()
+        self.connection = self.database.connect()
+
+    def query_log_and_execute(self, constructed_query):
+        """Logs and executes all queries ran in script"""
+        logging.info('SQL QUERY: %s', constructed_query)
+        print(constructed_query)
+        return self.connection.execute(constructed_query).fetchall()
+
+    def get_instruments_from_database(self):
+        """Retrieve current list of instruments from database"""
+        all_instruments = "SELECT id, name "\
+                          "FROM reduction_viewer_instrument"
+        return self.query_log_and_execute(all_instruments)
+
+    def rb_range_by_instrument(self, instrument, start_date, end_date):
+        """Retrieves run_numbers as longs for a given instrument
+        between two dates as a list of integers"""
+        missing_rb_calc_vars = {}
+
+        missing_rb_query = "SELECT run_number "\
+                           "FROM {} " \
+                           "WHERE instrument_id = {} " \
+                           "AND created " \
+                           "BETWEEN '{}' " \
+                           "AND '{}'".format(DatabaseMonitorChecks.table,
+                                             instrument,
+                                             start_date,
+                                             end_date)
+        # Execute query and append to dictionary
+        missing_rb_calc_vars['run_numbers'] = self.query_log_and_execute(missing_rb_query)
+
+        # Converts list of run number sets containing longs into list of integers
+        # [(123L), (456L)] -> [123, 456]
+        return [int(elem) for elem in list(itertools.chain.from_iterable(missing_rb_calc_vars['run_numbers']))]  # pylint: disable=line-too-long
```

Comment:
I would normally split this line up a little as it is fairly complex, however I think the comment describes it well enough to not require this 👍